lib/api.js

```diff
@@ -25,49 +24,50 @@ class GitLab {
     return Boolean(this.strictSSL)
   }
 
-  login (username, password, email) {
-    const body = {
-      login: username,
-      password: password
+  async projectInfo (token, org, repo) {
+    try {
+      let returnData = {};
+      const request = await got(`${this.url}/api/v4/projects?search=${repo}`, {
+        json: true,
+        headers: {
+          'Private-Token': token || process.env.GITLAB_ACCESS_TOKEN,
+        },
+      });
+      const repos = request.body || [];
+      if (repos.length === 0) { returnData = null; }
+      for (let i = 0; i < repos.length; i++) {
+        if (repos[i].name === repo && repos[i].namespace && repos[i].namespace.name === org) {
+          returnData = repos[i];
+        }
+      }
+      return returnData;
+    } catch (e) {
+      console.error(e);
     }
-    if (email) body.email = email
-    return got.post(this.url + '/api/v3/session', {
-      json: true,
-      body: body,
-      rejectUnauthorized: this.rejectUnauthorized()
-    }).then((res) => {
-      return res && res.body && res.body.private_token
-    })
   }
 
-  projectInfo (token, org, repo) {
-    return got(this.url + '/api/v3/projects?search=' + repo, {
-      json: true,
-      headers: {
-        'PRIVATE-TOKEN': token
-      },
-      rejectUnauthorized: this.rejectUnauthorized()
-    }).then((res) => {
-      let array = [].concat(res && res.body)
-      if (!array.length) return null
-      for (let i = 0, len = array.length, p; i < len; i++) {
-        p = array[i]
-        if (p && p.name === repo && p.namespace && p.namespace.name === org) return p
-      }
-      return null
-    })
+  async user() {
+    try {
+      const request = await got(`${this.url}/api/v4/user`, {
+        json: true,
+        headers: {
+          'Private-Token': this.token,
```

Comment:
`this.token` is no longer being set bc we are not using a global access token with sudo / admin perms. Instead lets just pass the token in as an arg.